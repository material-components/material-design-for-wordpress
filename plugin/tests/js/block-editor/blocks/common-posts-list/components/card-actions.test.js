/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import CardActions from '../../../../../../assets/src/block-editor/blocks/common-posts-list/components/card-actions';

/**
 * Render the component.
 *
 * @param {Object} props - Component props.
 * @return {Function} A functional component.
 */
const setup = props => {
	return render( <CardActions { ...props } /> );
};

const post = {
	authorDisplayName: 'Test user',
	commentsCount: 1,
};

describe( 'CardActions', () => {
	it( 'matches snapshot when the post author and comments count should be displayed', () => {
		const wrapper = setup( {
			displayPostAuthor: true,
			displayCommentsCount: true,
			post,
		} );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot when the post author and comments count should not be displayed', () => {
		const wrapper = setup( {
			displayPostAuthor: false,
			displayCommentsCount: false,
			post,
		} );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot when only the post author should be displayed', () => {
		const wrapper = setup( {
			displayPostAuthor: true,
			displayCommentsCount: false,
			post,
		} );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot when only the comments count should be displayed', () => {
		const wrapper = setup( {
			displayPostAuthor: false,
			displayCommentsCount: true,
			post,
		} );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot when the comments count is zero', () => {
		post.commentsCount = 0;
		const wrapper = setup( {
			displayPostAuthor: true,
			displayCommentsCount: true,
			post,
		} );
		expect( wrapper ).toMatchSnapshot();
	} );

	it( 'matches snapshot when the comments count is more than one', () => {
		post.commentsCount = 2;
		const wrapper = setup( {
			displayPostAuthor: true,
			displayCommentsCount: true,
			post,
		} );
		expect( wrapper ).toMatchSnapshot();
	} );
} );
