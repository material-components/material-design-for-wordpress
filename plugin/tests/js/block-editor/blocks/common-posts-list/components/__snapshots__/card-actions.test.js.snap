// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardActions matches snapshot when only the comments count should be displayed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mdc-card__actions"
      >
        <div
          class="mdc-card__action-buttons"
        >
          <button
            class="mdc-button mdc-card__action mdc-card__action--button mdc-button__comment-count"
          >
            <span
              class="mdc-button__ripple"
            />
            <i
              aria-hidden="true"
              class="material-icons mdc-button__icon"
            >
              comment
            </i>
            <span
              class="mdc-button__label"
            >
              1
            </span>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mdc-card__actions"
    >
      <div
        class="mdc-card__action-buttons"
      >
        <button
          class="mdc-button mdc-card__action mdc-card__action--button mdc-button__comment-count"
        >
          <span
            class="mdc-button__ripple"
          />
          <i
            aria-hidden="true"
            class="material-icons mdc-button__icon"
          >
            comment
          </i>
          <span
            class="mdc-button__label"
          >
            1
          </span>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CardActions matches snapshot when only the post author should be displayed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mdc-card__actions"
      >
        <div
          class="mdc-card__action-buttons"
        >
          <button
            class="mdc-button mdc-card__action mdc-card__action--button mdc-button__post-author"
          >
            <span
              class="mdc-button__ripple"
            />
            <i
              aria-hidden="true"
              class="material-icons mdc-button__icon"
            >
              face
            </i>
            <span
              class="mdc-button__label"
            >
              Test user
            </span>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mdc-card__actions"
    >
      <div
        class="mdc-card__action-buttons"
      >
        <button
          class="mdc-button mdc-card__action mdc-card__action--button mdc-button__post-author"
        >
          <span
            class="mdc-button__ripple"
          />
          <i
            aria-hidden="true"
            class="material-icons mdc-button__icon"
          >
            face
          </i>
          <span
            class="mdc-button__label"
          >
            Test user
          </span>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CardActions matches snapshot when the comments count is more than one 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mdc-card__actions"
      >
        <div
          class="mdc-card__action-buttons"
        >
          <button
            class="mdc-button mdc-card__action mdc-card__action--button mdc-button__post-author"
          >
            <span
              class="mdc-button__ripple"
            />
            <i
              aria-hidden="true"
              class="material-icons mdc-button__icon"
            >
              face
            </i>
            <span
              class="mdc-button__label"
            >
              Test user
            </span>
          </button>
          <button
            class="mdc-button mdc-card__action mdc-card__action--button mdc-button__comment-count"
          >
            <span
              class="mdc-button__ripple"
            />
            <i
              aria-hidden="true"
              class="material-icons mdc-button__icon"
            >
              comment
            </i>
            <span
              class="mdc-button__label"
            >
              2
            </span>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mdc-card__actions"
    >
      <div
        class="mdc-card__action-buttons"
      >
        <button
          class="mdc-button mdc-card__action mdc-card__action--button mdc-button__post-author"
        >
          <span
            class="mdc-button__ripple"
          />
          <i
            aria-hidden="true"
            class="material-icons mdc-button__icon"
          >
            face
          </i>
          <span
            class="mdc-button__label"
          >
            Test user
          </span>
        </button>
        <button
          class="mdc-button mdc-card__action mdc-card__action--button mdc-button__comment-count"
        >
          <span
            class="mdc-button__ripple"
          />
          <i
            aria-hidden="true"
            class="material-icons mdc-button__icon"
          >
            comment
          </i>
          <span
            class="mdc-button__label"
          >
            2
          </span>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CardActions matches snapshot when the comments count is zero 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mdc-card__actions"
      >
        <div
          class="mdc-card__action-buttons"
        >
          <button
            class="mdc-button mdc-card__action mdc-card__action--button mdc-button__post-author"
          >
            <span
              class="mdc-button__ripple"
            />
            <i
              aria-hidden="true"
              class="material-icons mdc-button__icon"
            >
              face
            </i>
            <span
              class="mdc-button__label"
            >
              Test user
            </span>
          </button>
          <button
            class="mdc-button mdc-card__action mdc-card__action--button mdc-button__comment-count"
          >
            <span
              class="mdc-button__ripple"
            />
            <i
              aria-hidden="true"
              class="material-icons mdc-button__icon"
            >
              comment
            </i>
            <span
              class="mdc-button__label"
            >
              0
            </span>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mdc-card__actions"
    >
      <div
        class="mdc-card__action-buttons"
      >
        <button
          class="mdc-button mdc-card__action mdc-card__action--button mdc-button__post-author"
        >
          <span
            class="mdc-button__ripple"
          />
          <i
            aria-hidden="true"
            class="material-icons mdc-button__icon"
          >
            face
          </i>
          <span
            class="mdc-button__label"
          >
            Test user
          </span>
        </button>
        <button
          class="mdc-button mdc-card__action mdc-card__action--button mdc-button__comment-count"
        >
          <span
            class="mdc-button__ripple"
          />
          <i
            aria-hidden="true"
            class="material-icons mdc-button__icon"
          >
            comment
          </i>
          <span
            class="mdc-button__label"
          >
            0
          </span>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CardActions matches snapshot when the post author and comments count should be displayed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mdc-card__actions"
      >
        <div
          class="mdc-card__action-buttons"
        >
          <button
            class="mdc-button mdc-card__action mdc-card__action--button mdc-button__post-author"
          >
            <span
              class="mdc-button__ripple"
            />
            <i
              aria-hidden="true"
              class="material-icons mdc-button__icon"
            >
              face
            </i>
            <span
              class="mdc-button__label"
            >
              Test user
            </span>
          </button>
          <button
            class="mdc-button mdc-card__action mdc-card__action--button mdc-button__comment-count"
          >
            <span
              class="mdc-button__ripple"
            />
            <i
              aria-hidden="true"
              class="material-icons mdc-button__icon"
            >
              comment
            </i>
            <span
              class="mdc-button__label"
            >
              1
            </span>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mdc-card__actions"
    >
      <div
        class="mdc-card__action-buttons"
      >
        <button
          class="mdc-button mdc-card__action mdc-card__action--button mdc-button__post-author"
        >
          <span
            class="mdc-button__ripple"
          />
          <i
            aria-hidden="true"
            class="material-icons mdc-button__icon"
          >
            face
          </i>
          <span
            class="mdc-button__label"
          >
            Test user
          </span>
        </button>
        <button
          class="mdc-button mdc-card__action mdc-card__action--button mdc-button__comment-count"
        >
          <span
            class="mdc-button__ripple"
          />
          <i
            aria-hidden="true"
            class="material-icons mdc-button__icon"
          >
            comment
          </i>
          <span
            class="mdc-button__label"
          >
            1
          </span>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CardActions matches snapshot when the post author and comments count should not be displayed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mdc-card__actions"
      >
        <div
          class="mdc-card__action-buttons"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mdc-card__actions"
    >
      <div
        class="mdc-card__action-buttons"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
